engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java

```diff
@@ -143,6 +145,22 @@ private Either<Failure, String> evalCandidateUsersExp(
         .map(ExpressionTransformer::asListLiteral);
   }
 
+  private Either<Failure, String> evalDateExp(final Expression date, final long scopeKey) {
+    if (date == null) {
+      return Either.right(null);
+    }
+    return expressionBehavior
+        .evaluateDateTimeExpression(date, scopeKey, true)
+        .flatMap(
```

Comment:
:wrench: Since we are returning only `Right` here, we can convert `flatMap` into `map` to make it look more functional as such: `.map(optionalDate -> optionalDate.map(ZonedDateTime::toString).orElse(null));`.

Reply:
Nice! Thanks for the suggestion.